From 40090eca032d602c78223abaa13ca4702db86b74 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Sun, 24 Aug 2008 17:44:10 +0200
Subject: [PATCH] Make RandRQueryVersion return 1.1 for swapped clients

Don't pretend we support randr 1.2 when queried by swapped clients.
The dispatch code for RandR 1.2 requests would return BadImplementation
anyway.
---
 randr/rrdispatch.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/randr/rrdispatch.c b/randr/rrdispatch.c
index 5525427..83deb6b 100644
--- a/randr/rrdispatch.c
+++ b/randr/rrdispatch.c
@@ -55,6 +55,11 @@ ProcRRQueryVersion (ClientPtr client)
     rep.majorVersion = SERVER_RANDR_MAJOR;
     rep.minorVersion = SERVER_RANDR_MINOR;
     if (client->swapped) {
+	/*
+	 * We return BadImplementation for swapped clients on 1.2 requests,
+	 * let's not do that
+	 */
+	rep.minorVersion = 1;
     	swaps(&rep.sequenceNumber, n);
     	swapl(&rep.length, n);
 	swapl(&rep.majorVersion, n);
-- 
1.5.6.3

